include/deal.II/base/function_parser.h

```diff
@@ -211,6 +211,11 @@ class Vector;
  *                        constants);
  * @endcode
  *
+ * @note The difference between this class and the SymbolicFunction class is
+ * that the SymbolicFunction class allows to compute first and second order
+ * derivatives (in a symbolic way), while this class computes first order
+ * derivatives only, using finite difference. For complicated expressions,
+ * this class is generally faster than SymbolicFunction.
```

Comment:
Out of curiosity, do you notice a significant performance difference between the two for your use cases?

Reply:
Not really. They are substantially identical, but I have not tested complicated stuff.